relay-dynamic-config/src/feature.rs

```diff
@@ -169,6 +169,23 @@ impl<'de> Deserialize<'de> for FeatureSet {
     }
 }
 
+impl Serialize for FeatureSet {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
Not a fan of this, I think we should do it either on deserialization *and* for every constructor (probably enough to intrade a new/from which merges the feature flags in).

Ideally all of the API knows of these 'magic' flags. E.g. `has(GRADUATED_FLAG)` returns true and not just on the next layer.

Reply:
The point is that these flags should not even be accessible through code anymore. I.e., `has(GRADUATED_FLAG)` does not return true, but it does not return false either, because you cannot (and should not) even ask for it. That's why we renamed them to `Deprecated*` in the past.

Reply:
That's okay, but instead of messing with the serializer I'd just explicitly manage them in the `FeatureSet`.

> but it does not return false either, because you cannot (and should not) even ask for it.

What happens if someone re-introduces a graduated feature flag? For processing Relays the feature will be false and downstream Relays think it's on.